src/lightning_app/utilities/proxies.py

```diff
@@ -372,20 +392,25 @@ def run_once(self):
 
         # 8. Deepcopy the work state and send the first `RUNNING` status delta to the flow.
         state = deepcopy(self.work.state)
+
+        # 9. Inform the flow the work is running and add the delta to the deepcopy.
         self.work._calls["latest_call_hash"] = call_hash
         self.work._calls[call_hash]["statuses"].append(make_status(WorkStageStatus.RUNNING))
-        self.delta_queue.put(ComponentDelta(id=self.work_name, delta=Delta(DeepDiff(state, self.work.state))))
+        delta = Delta(DeepDiff(state, self.work.state))
+        self.delta_queue.put(ComponentDelta(id=self.work_name, delta=delta))
 
-        # 9. Start the state observer thread. It will look for state changes and send them back to the Flow
-        # The observer has to be initialized here, after the set_state call above so that the thread can start with
-        # the proper initial state of the work
-        self.state_observer.start()
+        state += delta
```

Comment:
I'm not sure I'm following. Why do we have to add `delta` to `state` here? `state` is a deep copy of `self.work.state` and `delta` was obtained by diffing `state` and `self.work.state` immediately after.

If we need to do this here (i.e. if we see deltas accumulating after 2 lines of code that we need to take into account) it probably means we need to find a different ordering for the operations.

Reply:
We need to add the delta as we modified the state and this one is used to make a comparison at the end for the final delta. We ended up sending running status twice as this delta wasn't added.
This is mainly to avoid making another deepcopy as adding a delta is cheaper.

Reply:
Sorry where do we modify the state between 394 and 399?

Reply:
Removed it. It wasn't required.